### PR TITLE
[LAD Table/Table Sets] Configurable hidden columns

### LINKED ISSUE #6385: Allow view customization of LAD table columns
**Is your feature request related to a problem? Please describe.**
Users would like to be able to suppress columns in the same way they can in Telemetry Tables. In particular, allow the ability to suppress the timestamp column.

**Describe the solution you'd like**
Add an inspector view for edit mode on lad tables allowing you to hide "timestamp" and "unit" columns.

**Describe alternatives you've considered**
A mishmash of alphanumerics in display layouts.

**Additional context**
This would be the start of adding more configurability to LAD Tables and LAD Table Sets.

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #6385

### Describe your changes:

Added `configuration` to LAD Table and LAD Table Set domain objects. This is just the start of LAD Table configurability. I've modeled it after Telemetry Tables configurability. This PR only adds the ability to hide the "timestamp" and "unit" columns, but it lays the ground work for further configuration enhancements.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/LADTable/LADTableConfiguration.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        return configuration;
+    }
+
+    updateConfiguration(configuration) {
+        this.openmct.objects.mutate(this.domainObject, 'configuration', configuration);
+    }
+
+    objectMutated(configuration) {
+        if (configuration !== undefined) {
```
Comment (scottbell) [inline src/plugins/LADTable/LADTableConfiguration.js:48]: Could the configuration ever be `null` here too?

File: src/plugins/LADTable/LadTableSetView.js [hunk 2]
```diff
@@ -1,5 +1,27 @@
-import LadTableSet from './components/LadTableSet.vue';
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2022, United States Government
```
Comment (scottbell) [inline src/plugins/LADTable/LadTableSetView.js:2]: 2023! 😀

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 3]
```diff
@@ -0,0 +1,212 @@
+<template>
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:1]: need copyright preamble here

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        },
+        addLadTable(domainObject) {
+            let ladTable = {};
+            ladTable.domainObject = domainObject;
+            ladTable.key = this.openmct.objects.makeKeyString(domainObject.identifier);
+
+            this.$set(this.ladTelemetryObjects, ladTable.key, []);
+            this.ladTableObjects.push(ladTable);
+
+            let composition = this.openmct.composition.get(ladTable.domainObject);
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:108]: can these variables be `const` instead?

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            this.composition.off('remove', this.removeLadTable);
+            this.compositions.forEach(c => {
+                c.composition.off('add', c.addCallback);
+                c.composition.off('remove', c.removeCallback);
+            });
+        }
+    },
+    methods: {
+        addItem(domainObject) {
+            let item = {};
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:84]: I think we can make this `const`

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+            this.compositions.push({
+                composition,
+                addCallback,
+                removeCallback
+            });
+
+            this.shouldShowUnitsCheckbox();
+        },
+        removeLadTable(identifier) {
+            let index = this.ladTableObjects.findIndex(ladTable => this.openmct.objects.makeKeyString(identifier) === ladTable.key);
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:125]: `const` again

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            let ladTable = this.ladTableObjects[index];
+
+            this.$delete(this.ladTelemetryObjects, ladTable.key);
+            this.ladTableObjects.splice(index, 1);
+
+            this.shouldShowUnitsCheckbox();
+        },
+        addTelemetryObject(ladTable) {
+            return (domainObject) => {
+                let telemetryObject = {};
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:135]: `const`

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+                this.$set(this.ladTelemetryObjects, ladTable.key, telemetryObjects);
+
+                this.shouldShowUnitsCheckbox();
+            };
+        },
+        removeTelemetryObject(ladTable) {
+            return (identifier) => {
+                const keystring = this.openmct.objects.makeKeyString(identifier);
+                const telemetryObjects = this.ladTelemetryObjects[ladTable.key];
+                let index = telemetryObjects.findIndex(telemetryObject => keystring === telemetryObject.key);
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:151]: `const`

File: src/plugins/LADTable/components/LADTableConfiguration.vue [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+</template>
+
+<script>
+
+export default {
+    inject: ['openmct', 'ladTableConfiguration'],
+    data() {
+        return {
+            headers: {
+                timestamp: 'timestamp',
```
Comment (scottbell) [inline src/plugins/LADTable/components/LADTableConfiguration.vue:40]: We should capitalize these I think to match the other property headers
image

**Review by scottbell — CHANGES_REQUESTED** [review]
Nice work @jvigliotta! It works great:

https://user-images.githubusercontent.com/9853862/223044794-83b76d76-ce30-41f3-b671-3deee2a9a7ed.mov

I had some small cleanup changes, but otherwise works well. Can we add an e2e test too please?

File: src/plugins/LADTable/components/LADTableConfiguration.vue (hunk 9 above)
Comment (jvigliotta) [reply src/plugins/LADTable/components/LADTableConfiguration.vue:40]: Good call!

**Review by charlesh88 — APPROVED** [review]
"This is a thing of beauty" he said, shedding a small tear. LGTM!

File: src/plugins/LADTable/LADTableConfiguration.js (hunk 1 above)
Comment (jvigliotta) [reply src/plugins/LADTable/LADTableConfiguration.js:48]: I can't think of a situation where this would be `null`.

**Review by scottbell — CHANGES_REQUESTED** [review]
Code looks great - and excellent tests! Ran into one hiccup trying to navgiating back and forth in the property tabs:

https://user-images.githubusercontent.com/9853862/224243561-0538be93-b3b5-48c5-9691-a56d6ef09df5.mov

If I had to guess, it's the `isEditing` not being initialized properly when returning to the tab?

**Review by scottbell — APPROVED** [review]

**Pull request merged**